### ISSUE TITLE
Allow overriding JSON_THROW to something else than abort()

### ISSUE THREAD

**Issue by kaidokert** [nlohmann/json#938]
**Feature Request**

- Describe the feature in as much detail as possible.

On an embedded/rtos system, the stack is compiled without rtti/exceptions, and I'm running http workers in a separate thread. I can deal with failures by basically tearing the thread down and restarting it, however, i don't want to call abort() specifically and unconditionally. Also it's nice to still be able to obtain the error code from the exception

- Include sample usage where appropriate.

passing a -DJSON_THROW=my_deal_with_json_error(x) at compile time would allow easy customization of how the error handling gets dealt with

The addition is as simple as
```
#if !defined(JSON_THROW)
#define JSON_THROW abort
#endif
```

**Comment by nlohmann** [comment]
That seems like a reasonable extension, but maybe also for `JSON_TRY` and `JSON_CATCH`. What do you think?

**Comment by kaidokert** [comment]
I thought about that and didn't immediately think of a valid use case, but maybe doing something like
```
#define JSON_TRY(x) my_save_last_try_point(..., ___FILE__, ___LINE__ )
```
could be useful in debugging scenarios. I'll add and make a PR in a bit

**Comment by nlohmann** [comment]
Fixed with merged PR #940.
